src/main/java/com/ostapula/maven/sort/Person.java

```diff
@@ -38,7 +40,12 @@ public int compareTo(Person p) {
 		int result = compare(this.name, p.getName());
 		if (result != 0)
 			return result;
-		return compare(this.surname, p.getSurname());
+		result = compare(this.surname, p.getSurname());
+		if (result != 0)
+			return result;
+		if (this.equals(p))
```

Comment:
Consider that `result == 0` at this point (just before calling `equals()`). Is the call to `equals()` necessary here then?

Reply:
No, it is not necessary.